face-detection-explainer.md

```diff
@@ -12,66 +12,129 @@
+```js
+partial dictionary VideoFrameMetadata {
+  sequence<HumanFace> humanFaces;
+};
 
-The field `contour` provides an arbitrary number of points which enclose the face. Sophisticated implementations could provide a large number of points but initial implementations are expected to provide four contour points located in the corners of a rectangular region which describes the face bounding box. User agent is allowed to provide a minimum of one point, in which case the point should be located at the center of the face.
+dictionary HumanFace {
+  long              id;
+  float             probability;
+  DOMRectReadOnly   boundingBox;
+  HumanFaceLandmark leftEye;
+  HumanFaceLandmark rightEye;
+  HumanFaceLandmark mouth;
+};
 
-The field `landmarks` provides a list of facial features belonging to the detected face, such as eyes or mouth. User agent may return accurate contour for the features, but early implementations are expected to deliver only a single point corresponding to the center of a feature.
+dictionary HumanFaceLandmark {
+  Point2D           centerPoint;
+};
 
-The contour of the detected faces and landmarks is described as a sequence of `Point2D` and the units
-conform to the 
+```
+
+### Constraints
+
+The second part consists of constraints to `getUserMedia()` or `applyConstrains()` which are used to negotiate and enable the desired face detection mode. Often, camera drivers run already internally face detection for 3A, so enabling face detection might just make the results available to Web applications. The corresponding members are added also to media track settings and capabilities. 
+
+```js
+partial dictionary MediaTrackSupportedConstraints {
+  boolean humanFaceDetectionMode = true;
+  boolean humanFaceLandmarkDetectionMode = true;
+};
+
+partial dictionary MediaTrackCapabilities {
+  sequence<DOMString> humanFaceDetectionMode;
+  sequence<DOMString> humanFaceLandmarkDetectionMode;
+};
+
+partial dictionary MediaTrackConstraintSet {
+  ConstrainDOMString humanFaceDetectionMode;
+  ConstrainDOMString humanFaceLandmarkDetectionMode;
+};
+
+partial dictionary MediaTrackSettings {
+  DOMString humanFaceDetectionMode;
+  DOMString humanFaceLandmarkDetectionMode;
+};
+
+enum HumanFaceDetectionMode {
+  "none",          // Face or landmark detection is not needed
+  "bounding-box",  // Bounding box of the detected object is returned

```

Comment:
WebIDL is for specs not explainers IMHO. Can this be replaced with JS snippets showing expected syntax?

if not then please replace `js` with `idl` on the first line to get the correct syntax highlighting.

Reply:
I like having WebIDL in explainers as long as we're not yet ready to land it in specs. It's a pain to have it both places.

Reply:
I think JS snippets are better for the examples at the end of the explainer. For explaining things, I found JS to be inconvenient. How would an enum be written in Javascript, for instance? Therefore, replaced js with idl.